python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -239,32 +247,35 @@ def _address_family_evpn(self: AvdStructuredConfigOverlay) -> dict | None:
                 }
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
-            # TODO: - assess this condition
+            # TODO: - assess this condition - both can't be true at the same time.
             if self.shared_utils.overlay_evpn_mpls is True and self.shared_utils.overlay_evpn_vxlan is not True:
-                overlay_peer_group_name = self.shared_utils.bgp_peer_groups["mpls_overlay_peers"]["name"]
-                peer_groups.append({"name": overlay_peer_group_name, "activate": True})
+                overlay_peer_group = {"name": self._get_peer_group_name("mpls_overlay_peers"), "activate": True}
                 address_family_evpn["neighbor_default"] = {"encapsulation": "mpls"}
                 if self.shared_utils.overlay_ler is True:
                     address_family_evpn["neighbor_default"]["next_hop_self_source_interface"] = "Loopback0"
 
                 if self._is_mpls_server is True:
-                    peer_groups.append({"name": self.shared_utils.bgp_peer_groups["rr_overlay_peers"]["name"], "activate": True})
-
-            # partly duplicate with ebgp
-            if (
-                self.shared_utils.overlay_vtep is True
-                and self.shared_utils.evpn_role != "server"
-                and (peer_group := get_item(peer_groups, "name", overlay_peer_group_name)) is not None
-            ):
-                peer_group.update(
+                    peer_groups.append({"name": self._get_peer_group_name("rr_overlay_peers"), "activate": True})
+
+            if self.shared_utils.overlay_vtep is True and self.shared_utils.evpn_role != "server" and overlay_peer_group:
+                overlay_peer_group.update(
                     {
                         "route_map_in": "RM-EVPN-SOO-IN",
                         "route_map_out": "RM-EVPN-SOO-OUT",
                     },
                 )
 
             if self._is_wan_server_with_peers:
-                peer_groups.append({"name": self.shared_utils.bgp_peer_groups["wan_rr_overlay_peers"]["name"], "activate": True})
+                peer_groups.append(
+                    {
+                        "name": self._get_peer_group_name("wan_rr_overlay_peers"),
+                        "activate": True,
+                        "encapsulation": self.shared_utils.wan_encapsulation,
+                    }
+                )
+
+        if overlay_peer_group:
+            peer_groups.append(overlay_peer_group)
```

Comment:
The reordering structured config is because of this change. Why is this better than the two separate appends?

Reply:
needed since our update of route-map is done in a sane way now instead of retrieving the peer-group back from the list and updating it.